src/components/welcome/MyAccounts/index.tsx

```diff
@@ -30,7 +32,19 @@ const AccountsList = ({ safes, onLinkClick }: AccountsListProps) => {
           <CreateButton />
         </Box>
 
-        <PaginatedSafeList title="My accounts" safes={ownedSafes} onLinkClick={onLinkClick} />
+        <PaginatedSafeList
+          title="My accounts"
+          safes={ownedSafes}
+          onLinkClick={onLinkClick}
+          noSafesMessage={
+            <>
+              Connect a wallet to view your Safe Accounts or to create a new one
+              <Button onClick={handleConnect} disableElevation size="small" sx={{ mt: 2 }}>
+                Connect a wallet
+              </Button>
```

Comment:
There's a `ConnectWallet` button component, it can be extended to allow custom text.